mm2src/docker_tests.rs

```diff
@@ -899,12 +902,142 @@ mod docker_tests {
         // the result of equation x + x / 777 + 0.00002 = 1
         assert_eq!(json["result"]["numer"], Json::from("38849223"));
         assert_eq!(json["result"]["denom"], Json::from("38900000"));
+
+        let rc = unwrap!(block_on(mm_alice.rpc(json! ({
+            "userpass": mm_alice.userpass,
+            "method": "sell",
+            "base": "MYCOIN1",
+            "rel": "MYCOIN",
+            "price": 1,
+            "volume": {
+                "numer": json["result"]["numer"],
+                "denom": json["result"]["denom"],
+            }
+        }))));
+        assert!(rc.0.is_success(), "!sell: {}", rc.1);
+
+        unwrap!(block_on(mm_alice.stop()));
+    }
+
+    // https://github.com/KomodoPlatform/atomicDEX-API/issues/733
+    #[test]
+    fn test_get_max_taker_vol_dex_fee_threshold() {
+        let (_ctx, _, alice_priv_key) = generate_coin_with_random_privkey("MYCOIN1", "0.05328455".parse().unwrap());
+        let coins = json! ([
+            {"coin":"MYCOIN","asset":"MYCOIN","txversion":4,"overwintered":1,"txfee":10000,"protocol":{"type":"UTXO"}},
+            {"coin":"MYCOIN1","asset":"MYCOIN1","txversion":4,"overwintered":1,"txfee":10000,"protocol":{"type":"UTXO"}},
+        ]);
+        let mut mm_alice = unwrap!(MarketMakerIt::start(
+            json! ({
+                "gui": "nogui",
+                "netid": 9000,
+                "dht": "on",  // Enable DHT without delay.
+                "passphrase": format!("0x{}", hex::encode(alice_priv_key)),
+                "coins": coins,
+                "rpc_password": "pass",
+                "i_am_see": true,
+            }),
+            "pass".to_string(),
+            None,
+        ));
+        let (_alice_dump_log, _alice_dump_dashboard) = mm_dump(&mm_alice.log_path);
+        unwrap!(block_on(
+            mm_alice.wait_for_log(22., |log| log.contains(">>>>>>>>> DEX stats "))
+        ));
+
+        log!([block_on(enable_native(&mm_alice, "MYCOIN1", vec![]))]);
+        log!([block_on(enable_native(&mm_alice, "MYCOIN", vec![]))]);
+        let rc = unwrap!(block_on(mm_alice.rpc(json! ({
+            "userpass": mm_alice.userpass,
+            "method": "max_taker_vol",
+            "coin": "MYCOIN1",
+        }))));
+        assert!(rc.0.is_success(), "!max_taker_vol: {}", rc.1);
+        let json: Json = json::from_str(&rc.1).unwrap();
+        // the result of equation x + 0.0001 + 0.00002 = 0.05328455
```

Comment:
I've got 0.05316455 result of the comment equation, but numer/denom is 0.05298455. I guess I misunderstand something, please correct me

Reply:
Yea, there's a mistake in comment, let me fix it.
The txfee is configured as 10k satoshis in this test so actual equation is `x + 0.0001 (dex fee) + 0.0002 (miner fee 0.0001 * 2) = 0.05328455` which solution is `0.05298455`

Reply:
Fixed, thanks a lot for your review!